### ISSUE TITLE
ISO 8601 Date parse error

### ISSUE THREAD

**Issue by category** [cognitect-labs/aws-api#144]
### Description

This exception was observed when using `[com.cognitect.aws/kafka "801.2.686.0"]` to make a `{:op :ListConfigurations}` request:
```
   {:request
    ...
     {:op :ListConfigurations}),
    :response
    {:cognitect.anomalies/category :cognitect.anomalies/fault,
     :cognitect.aws.client/throwable #error {
    :cause "Unparseable date: \"2020-07-06T08:58:13.416Z\""
```

At the line in question:

[4537c9cb242ce7ae4720a1fc789cb8757e11ca35](https://github.com/cognitect-labs/aws-api/blob/4537c9cb242ce7ae4720a1fc789cb8757e11ca35/src/cognitect/aws/shape.clj#L83)

When the cognitect client parses a JSON response body from the AWS API, if the timestamp format of the date field is marked as ISO 8601 but does not have the format `yyyy-MM-dd'T'HH:mm:ssXXX`, then a `java.textParseException` exception is thrown:
```clojure
user> (cognitect.aws.shape/parse-date {:timestampFormat "iso8601"} "2020-07-06T10:59:13Z")
#inst "2020-07-06T10:59:13.000-00:00"
user> (cognitect.aws.shape/parse-date {:timestampFormat "iso8601"} "2020-07-06T10:59:13.417Z")
Execution error (ParseException) at java.text.DateFormat/parse (DateFormat.java:395).
Unparseable date: "2020-07-06T10:59:13.417Z"
```

### Solution
With the following change:
```diff
diff --git a/src/cognitect/aws/shape.clj b/src/cognitect/aws/shape.clj
--- a/src/cognitect/aws/shape.clj
+++ b/src/cognitect/aws/shape.clj
@@ -82,7 +82,11 @@
   [shape data]
   (condp = (:timestampFormat shape)
     "rfc822"  (util/parse-date util/rfc822-date-format data)
-    "iso8601" (util/parse-date util/iso8601-date-format data)
+    "iso8601" (->> [util/iso8601-date-format
+                    util/iso8601-msecs-date-format]
+                   (map #(try (util/parse-date % data) (catch java.text.ParseException _ nil)))
+                   (filter identity)
+                   first)
     (cond (int? data)
           (java.util.Date. (* 1000 ^int data))
           (double? data)
```

calling `cognitect.aws.shape/parse-date` with a timestamp string that has the format `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'` succeeds:
```clojure
user> (cognitect.aws.shape/parse-date {:timestampFormat "iso8601"} "2020-07-06T10:59:13.417Z")
#inst "2020-07-06T10:59:13.417-00:00"
```

**Comment by dchelimsky** [comment]
Fixed on master. Will be part of the next release (probably today).

Thanks for the well documented issue!

**Comment by dchelimsky** [comment]
Released in 0.8.469.
